app/cdap/components/GridTable/index.tsx

```diff
@@ -44,20 +54,24 @@ export default function GridTable() {
 
   const classes = useStyles();
 
-  const [loading, setLoading] = useState(false);
+  const [loading, setLoading] = useState<boolean>(false);
   const [headersNamesList, setHeadersNamesList] = useState<IHeaderNamesList[]>([]);
-  const [rowsDataList, setRowsDataList] = useState([]);
-  const [gridData, setGridData] = useState({} as IExecuteAPIResponse);
-  const [missingDataList, setMissingDataList] = useState([]);
-  const [workspaceName, setWorkspaceName] = useState('');
-  const [invalidCountArray, setInvalidCountArray] = useState([
+  const [rowsDataList, setRowsDataList] = useState<IRowData[]>([]);
+  const [gridData, setGridData] = useState<IExecuteAPIResponse>();
+  const [missingDataList, setMissingDataList] = useState<IMissingListData[]>([]);
+  const [workspaceName, setWorkspaceName] = useState<string>('');
+  const [invalidCountArray, setInvalidCountArray] = useState<Array<Record<string, string>>>([
     {
       label: 'Invalid',
       count: '0',
     },
   ]);
   const [showBreadCrumb, setShowBreadCrumb] = useState<boolean>(true);
-
+  const [directiveFunction, setDirectiveFunction] = useState<string>('');
```

Comment:
why do we have directive keyword here
plz explain the importance of the keyword "directive" here

otherwise use something specific to transformations